File: tests/fixtures/anti-patterns/typescript/unused-imports-vulnerable.ts
/**
 * Test Fixture: Unused Imports Vulnerable Pattern (TypeScript)
 *
 * This file contains import statements that are never used.
 * Expected detections: >=1 violation for AP-007
 * Rule ID: AP-007
 * Severity: MEDIUM (info)
 */

import { useState, useEffect } from 'react';  // useEffect UNUSED - VULNERABLE
import * as lodash from 'lodash';              // UNUSED - VULNERABLE
import axios from 'axios';                     // UNUSED - VULNERABLE

export function MyComponent(): void {
  const [count, setCount] = useState(0);  // Only useState is used
  setCount(count + 1);
}

File: tests/fixtures/anti-patterns/python/unused_imports_vulnerable.py
"""
Test Fixture: Unused Imports Vulnerable Pattern (Python)

This file contains import statements that are never used.
Expected detections: >=2 violations for AP-007
Rule ID: AP-007
Severity: MEDIUM (info)
"""

import os  # USED - referenced below
import sys  # UNUSED - VULNERABLE
import json  # UNUSED - VULNERABLE
import logging  # UNUSED - VULNERABLE
from typing import List, Dict, Optional  # Dict and Optional UNUSED - VULNERABLE
from pathlib import Path  # UNUSED - VULNERABLE


def get_current_directory():
    """Get current working directory."""
    return os.getcwd()  # Only 'os' is used


def process_items(items: List[str]) -> List[str]:
    """Process items - only List from typing is used."""
    return [item.upper() for item in items]
